site/documentation/content/api/command-and-control-pubsub/index.md

```diff
@@ -0,0 +1,178 @@
+and a *correlation-id* attribute, providing the identifier that is used to correlate a response message to the original
+request.
+
+**Preconditions**
+
+1. The topics `projects/${google_project_id}/topics/${tenant_id}.command`
+   and `projects/${google_project_id}/topics/${tenant_id}.command_response` exist
+2. The subscription `projects/${google_project_id}/subscriptions/${tenant_id}.command` to the
+   topic `projects/${google_project_id}/topics/${tenant_id}.command` exists.
+3. A subscription to the topic `projects/${google_project_id}/topics/${tenant_id}.command_response` exists.
+4. The *Business Application* is authorized to write to the `projects/${google_project_id}/topics/${tenant_id}.command`
+   topic and read from the subscriptions.
+
+**Message Flow**
+
+1. The *Business Application* writes a command message to the
+   `projects/${google_project_id}/topics/${tenant_id}.command` topic on *Pub/Sub*.
+2. Hono consumes the message from *Pub/Sub* and forwards it to the device, provided that the target device is
+   connected and is accepting commands.
+3. The device sends a command response message. Hono writes that message to
+   the `projects/${google_project_id}/topics/${tenant_id}.command_response` topic
+   on *Pub/Sub*.
+4. The *Business Application* consumes the command response message from an independently created subscription to
+   the `projects/${google_project_id}/topics/${tenant_id}.command_response` topic.
+
+**Command Message Format**
+
+Metadata MUST be set as Pub/Sub attributes on a message.
+The following table provides an overview of the attributes the *Business Application* needs to set on a request/response
+command message.
+
+| Name                                         | Mandatory | Type      | Description                                                                                                                                |
+|:---------------------------------------------|:---------:|:----------|:-------------------------------------------------------------------------------------------------------------------------------------------|
+| *correlation-id*                             |    yes    | *string*  | The identifier used to correlate a response message to the original request. It is used as the *correlation-id* attribute in the response. |
+| *deviceId*                                   |    yes    | *string*  | The identifier of the device that the command is targeted at.                                                                              |
+| *response-required*                          |    yes    | *boolean* | MUST be set with a value of `true`, meaning that the device is required to send a response for the command.                                |
+| *subject*                                    |    yes    | *string*  | The name of the command to be executed by the device.                                                                                      |
```

Comment:
can we also add an optional *content-type* attribute and *delivery-failure-notification-metadata* as defined by the Kafka based API?